packages/core/src/submodules/cbor/cbor-encode.ts

```diff
@@ -0,0 +1,250 @@
+          data[cursor++] = (major << 5) | extendedFloat32;
+          dataView.setUint32(cursor, value);
+          cursor += 4;
+        } else {
+          data[cursor++] = (major << 5) | extendedFloat64;
+          dataView.setBigUint64(cursor, BigInt(value));
+          cursor += 8;
+        }
+        continue;
+      }
+      data[cursor++] = (majorSpecial << 5) | extendedFloat64;
+      dataView.setFloat64(cursor, input);
+      cursor += 8;
+      continue;
+    } else if (typeof input === "bigint") {
+      const major = input >= 0 ? majorUint64 : majorNegativeInt64;
+      const value = input >= 0 ? input : -input - BigInt(1);
+      const n = Number(value);
+      if (n < 24) {
+        data[cursor++] = (major << 5) | n;
+      } else if (n < 256 /* 2 ** 8 */) {
+        data[cursor++] = (major << 5) | 24;
+        data[cursor++] = n;
+      } else if (n < 65536 /* 2 ** 16 */) {
+        data[cursor++] = (major << 5) | extendedFloat16;
+        data[cursor++] = n >> 8;
+        data[cursor++] = n & 0b1111_1111;
+      } else if (n < 4294967296 /* 2 ** 32 */) {
+        data[cursor++] = (major << 5) | extendedFloat32;
+        dataView.setUint32(cursor, n);
+        cursor += 4;
+      } else {
+        data[cursor++] = (major << 5) | extendedFloat64;
+        dataView.setBigUint64(cursor, value);
+        cursor += 8;
+      }
+      continue;
+    } else if (input === null) {
+      data[cursor++] = (majorSpecial << 5) | specialNull;
+      continue;
+    } else if (typeof input === "boolean") {
+      data[cursor++] = (majorSpecial << 5) | (input ? specialTrue : specialFalse);
+      continue;
+    } else if (typeof input === "undefined") {
+      // Note: Smithy spec requires that undefined not be serialized
+      // though the CBOR spec includes it.
+      throw new Error("@smithy/core/cbor: client may not serialize undefined value.");
+    } else if (Array.isArray(input)) {
+      for (let i = input.length - 1; i >= 0; --i) {
+        encodeStack.push(input[i]);
+      }
+      encodeStack.push({
+        headerEncode,
+        major: majorList,
+        count: input.length,
+      });
+      continue;
+    } else if (typeof input.byteLength === "number") {
+      ensureSpace(input.length * 2);
+      encodeHeader(majorUnstructuredByteString, input.length);
+      data.set(input, cursor);
+      cursor += input.byteLength;
+      continue;
+    } else if (typeof input === "object" && "tag" in input && "value" in input && Object.keys(input).length === 2) {
+      encodeStack.push(input.value);
+      encodeStack.push({
+        headerEncode,
+        major: majorTag,
+        count: input.tag,
+      });
+      continue;
+    } else if (typeof input === "object") {
+      const keys = Object.keys(input);
+      for (let i = keys.length - 1; i >= 0; --i) {
+        const key = keys[i];
+        encodeStack.push(input[key]);
+        encodeStack.push(key);
+      }
+      encodeStack.push({
```

Comment:
Good spot 😮. The push of a header encoding task always directly precedes the `continue` statement, so it gets popped immediately. 

Yes, these can be called on the spot instead of pushed.

Reply:
this changes improves relative serialization throughput from the 25-28% range to the 28-30% range for `list<struct> PutMetricData-like`.